libraries/kotlin.test/testng/src/test/kotlin/TestNGContributorTest.kt

```diff
@@ -0,0 +1,28 @@
+package kotlinx.testing.tests
+
+import org.testng.annotations.*
+import org.testng.*
+import java.util.concurrent.*
+
+class TestNGContributorTest {
+    @Test
+    fun smokeTest() {
+        Assert.assertEquals("TestNGAsserter", asserter.`class`.simpleName)
+    }
+
+    @Test
+    fun `should fail to contribute if it was run outside of testng`() {
+        @Suppress("PLATFORM_CLASS_MAPPED_TO_KOTLIN")
+        val q = ArrayBlockingQueue<java.lang.Object>(1)
+
+        Thread {
+            q.put(asserter)
+        }.start()
+
+        Assert.assertEquals("DefaultAsserter", q.take().`class`.simpleName)
+    }
+
+    @Suppress("PLATFORM_CLASS_MAPPED_TO_KOTLIN")
+    private val asserter: java.lang.Object
```

Comment:
`asserter` is a public top-level property, it can be obtained directly without reflection.